src/vs/workbench/parts/preferences/browser/preferencesEditor.ts

```diff
@@ -427,13 +474,17 @@ class PreferencesRenderers extends Disposable {
 		return preferencesRenderer ? (<ISettingsEditorModel>preferencesRenderer.preferencesModel).settingsGroups : [];
 	}
 
-	private _filterPreferences(filter: string, preferencesRenderer: IPreferencesRenderer<ISetting>): IFilterResult {
-		let filterResult = null;
+	private _filterPreferences(filter: string, searchModel: PreferencesSearchModel, preferencesRenderer: IPreferencesRenderer<ISetting>): TPromise<IFilterResult> {
 		if (preferencesRenderer) {
-			filterResult = filter ? (<ISettingsEditorModel>preferencesRenderer.preferencesModel).filterSettings(filter) : null;
-			preferencesRenderer.filterPreferences(filterResult);
```

Comment:
Why not preferences model is the single point of contact, instead of having a new model for search. Internally preferences model can use search model or what ever it wants?

Reply:
The "searchModel" isn't the same as the editor model that already exists. There's one search model which is the result of the search, which gets applied to the two preferences models. I need something like this so that the remote search isn't performed twice.